auth/auth-server/src/main.rs

```diff
@@ -82,19 +90,43 @@ async fn main() {
     let args = Cli::parse();
     let listen_addr: SocketAddr = ([0, 0, 0, 0], args.port).into();
 
+    // Setup logging
+    configure_telemetry(
+        args.datadog_logging, // datadog_enabled
+        false,                // otlp_enabled
+        false,                // metrics_enabled
+        "".to_string(),       // collector_endpoint
+        "",                   // statsd_host
+        0,                    // statsd_port
+    )
+    .expect("failed to setup telemetry");
+
     // Create the server
     let server = Server::new(args).await.expect("Failed to create server");
     let server = Arc::new(server);
 
-    // TODO: Setup logging
-
     // --- Routes --- //
 
     // Ping route
     let ping = warp::path("ping")
         .and(warp::get())
         .map(|| warp::reply::with_status("PONG", StatusCode::OK));
 
+    // Add an API key
+    let add_api_key = warp::path(API_KEYS_PATH)
+        .and(warp::post())
+        .and(warp::body::json())
+        .and(with_server(server.clone()))
+        .and_then(|request, server: Arc<Server>| async move { server.add_key(request).await });
+
+    // Expire an API key
+    let expire_api_key = warp::path(API_KEYS_PATH)
+        .and(warp::path::param::<Uuid>())
+        .and(warp::path(DEACTIVATE_API_KEY_PATH))
+        .and(warp::post())
+        .and(with_server(server.clone()))
+        .and_then(|id: Uuid, server: Arc<Server>| async move { server.expire_key(id).await });
```

Comment:
what's our plan for authenticating these endpoints? introduce another symmetric key and use HMAC auth like we do elsewhere? or am I mistaken in thinking we'll need these endpoints authenticated?

Reply:
that's correct, we'll have an API key for admin operations here.